### PR TITLE
Add BuildTemplate for makisu

### PR DESCRIPTION
**Pull request by johscheuer**
This PR adds a simple BuildTemplate for makisu (https://github.com/uber/makisu). As soon as https://github.com/knative/build/pull/550 gets merged I would adjust the Secret Volume to be rendered by a Template Parameter rather than a hardcoded string.

Currently the makisu setup doesn't use a global build cache (mentioned here: https://github.com/uber/makisu#using-cache) if you want I can add an example for the global cache too.

Fixes: https://github.com/knative/build-templates/issues/77

### CODE REVIEW

**Comment by johscheuer** [comment]
ping @ImJasonH  and @mattmoor

File: makisu/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+## Parameters
+
+* **IMAGE**: The Docker image name to apply to the newly built image.
+  (_required_)
+* **CONTEXTPATH**: The path to the build context (_default:_
+  `/workspace`)
+* **PUSH_REGISTRY**: The Registry to push the image to (_default:_
+  `index.docker.io`)
+* **REGISTRY_SECRET**: Secret containing information about the used regsitry (_default:_
+  `docker-registry-config`) **This Parameter currently doesn't work: https://github.com/knative/build/pull/550**
```
Comment (ImJasonH) [inline makisu/README.md]: This PR is merged now, available in the latest Knative Build release (0.4)

File: makisu/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+kubectl apply -f https://raw.githubusercontent.com/knative/build-templates/master/makisu/makisu.yaml
+```
+
+## Parameters
+
+* **IMAGE**: The Docker image name to apply to the newly built image.
+  (_required_)
+* **CONTEXTPATH**: The path to the build context (_default:_
+  `/workspace`)
+* **PUSH_REGISTRY**: The Registry to push the image to (_default:_
```
Comment (ImJasonH) [inline makisu/README.md]: How does this work if `IMAGE` is not a Dockerhub image? Is it just unused?

File: makisu/makisu.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  # ToDo currently not possible see: https://github.com/knative/build/pull/550
+  - name: REGISTRY_SECRET
+    description: Secret containing information about the used regsitry.
+    default: docker-registry-config
+  steps:
+  - name: build-and-push
+    image: gcr.io/makisu-project/makisu:v0.1.8
+    args:
+      - build
+      - --push=${PUSHREGISTRY}
```
Comment (ImJasonH) [inline makisu/makisu.yaml]: typo: `PUSH_REGISTRY`

File: makisu/makisu.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    env:
+    - name: DOCKER_CONFIG
+      value: /builder/home/.docker
+    volumeMounts:
+    - name: registry-config
+      mountPath: /registry-config
+  volumes:
+  - name: registry-config
+    secret:
+      # ToDo currently not possible ${REGISTRY_SECRET}
```
Comment (ImJasonH) [inline makisu/makisu.yaml]: This should be possible now.

File: makisu/README.md (hunk 1 above)
Comment (johscheuer) [reply makisu/README.md]: I adjusted it

File: makisu/README.md (hunk 2 above)
Comment (johscheuer) [reply makisu/README.md]: I added an example: https://github.com/knative/build-templates/pull/90/files#diff-b4a0ee2d59c0f7719473b6ba4d25fa06R66

Comment (johscheuer) [reply makisu/README.md]: You need to create a secret for the registry like in the docker example and then reference it with `PUSH_REGISTRY`, makisu also supports to push to multiple registries (which is currently not implemented in this PR.

File: makisu/README.md (hunk 1 above)
Comment (ImJasonH) [reply makisu/README.md]: It looks like this still says it doesn't work?

Comment (johscheuer) [reply makisu/README.md]: Good catch, thanks!

**Pull request merged**